### PR TITLE
Change the way args are passed to modules

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes spring-cloud/spring-cloud-dataflow-admin-cloudfoundry#56
Might fix spring-cloud/spring-cloud-stream-modules#195 :)

### CODE REVIEW

File: spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java [hunk 1]
```diff
@@ -317,28 +318,16 @@ private ModuleStatus notRunningInstancesStatus(ModuleDeploymentId id, CloudAppli
 	private Map<String, String> toEnvironmentVariables(HashMap<String, String> args) {
 
 		Map<String, String> env = new HashMap<>(args.size());
-		StringBuilder sb = new StringBuilder();
-		for (Map.Entry<String, String> entry : args.entrySet()) {
-			String oneArg = "--" + entry.getKey() + "=" + quoteValue(entry.getValue());
-			sb.append(bashEscape(oneArg)).append(' ');
+		ObjectMapper objectMapper = new ObjectMapper();
+		try {
+			env.put("SPRING_APPLICATION_JSON", objectMapper.writeValueAsString(args));
```
Comment (artembilan) [inline spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java]: @ericbottard , looks cool :+1: 

But would you mind to explain who reads this variable from the EVN and parse that JSON to the proper `PropertySource`?

Maybe some inline comment here would be good one to have...

Thanks in advance!

Comment (ericbottard) [inline spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java]: It's actually a boot feature: https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config

Comment (artembilan) [inline spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java]: Great! Thank you!

M-m-m. Does it really work as it is described:
> `$ SPRING_APPLICATION_JSON='{"foo":{"bar":"spam"}}' java -jar myapp.jar`

> In this example you will end up with foo.bar=spam in the Spring Environment.

I mean this your `Map -> JSON` conversion.
How does that `args` Map looks? 
And what is the output JSON?
Any samples, test?

Sorry, if I am annoying, but I wonder if we really should have something to confirm that we don't introduce an new issue...

Comment (ericbottard) [inline spring-cloud-dataflow-deployer-cloudfoundry/src/main/java/org/springframework/cloud/dataflow/module/deployer/cloudfoundry/ApplicationModuleDeployer.java]: The things this code puts in the map are always flat. They're read back on the other end as before and interpreted by boot, as before.
Agreed about tests, but I actually plan on adding a test in the TCK suite that validates that the deployer knows how to pass args, including ones with funky characters.

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Rebased and merged as 5a5be0b

**Comment by ilayaperumalg** [comment]
Also verified the changes by running a CF deployment successfully.

**Pull request closed without merging**